hydra-node/test/Hydra/NetworkSpec.hs

```diff
@@ -37,35 +37,35 @@ spec = describe "Networking layer" $ do
 
   describe "Ouroboros Network" $ do
     it "broadcasts messages to single connected peer" $ do
-      received <- newEmptyMVar
+      received <- newEmptyTMVarIO
       showLogsOnFailure $ \tracer -> failAfter 10 $ do
         withOuroborosNetwork tracer (lo, 45778) [(lo, 45779)] (const @_ @(HydraMessage SimpleTx) $ pure ()) $ \hn1 ->
-          withOuroborosNetwork @(HydraMessage SimpleTx) tracer (lo, 45779) [(lo, 45778)] (putMVar received) $ \_ -> do
+          withOuroborosNetwork @(HydraMessage SimpleTx) tracer (lo, 45779) [(lo, 45778)] (atomically . putTMVar received) $ \_ -> do
             broadcast hn1 requestTx
-            takeMVar received `shouldReturn` requestTx
+            atomically (takeTMVar received) `shouldReturn` requestTx
 
     it "broadcasts messages between 3 connected peers" $ do
-      node1received <- newEmptyMVar
-      node2received <- newEmptyMVar
-      node3received <- newEmptyMVar
+      node1received <- newEmptyTMVarIO
+      node2received <- newEmptyTMVarIO
+      node3received <- newEmptyTMVarIO
       showLogsOnFailure $ \tracer -> failAfter 10 $ do
-        withOuroborosNetwork @(HydraMessage SimpleTx) tracer (lo, 45678) [(lo, 45679), (lo, 45680)] (putMVar node1received) $ \hn1 ->
-          withOuroborosNetwork tracer (lo, 45679) [(lo, 45678), (lo, 45680)] (putMVar node2received) $ \hn2 -> do
-            withOuroborosNetwork tracer (lo, 45680) [(lo, 45678), (lo, 45679)] (putMVar node3received) $ \hn3 -> do
+        withOuroborosNetwork @(HydraMessage SimpleTx) tracer (lo, 45678) [(lo, 45679), (lo, 45680)] (atomically . putTMVar node1received) $ \hn1 ->
```

Comment:
No `MVar` in our prelude / do we have something against `MVar` for simple synchronization?

Reply:
There's no MVar in io-sim, I'd be curious to know why :thinking: